src/controllers/EvidenceUploadController.ts

```diff
@@ -34,32 +39,98 @@ export class EvidenceUploadController extends BaseController<OtherReason> {
         return appeal.reasons?.other;
     }
 
+    private async renderUploadError(appeal: Appeal, text: string): Promise<void> {
+        const that = this;
+        const validationResult: ValidationResult = new ValidationResult([
+            new ValidationError('file', text)
+        ]);
+
+        return await that.renderWithStatus(UNPROCESSABLE_ENTITY)(
+            that.template, {
+                ...this.prepareViewModelFromAppeal(appeal),
+                ...this.httpContext.request.body,
+                validationResult
+            }
+        );
+    }
+
     protected getExtraActionHandlers(): Record<string, FormActionHandler | FormActionHandlerConstructor> {
         const that = this;
+
+        const fileTooLargeError: string = 'File size must be smaller than 4MB';
+        const fileNotSupportedError: string = 'The selected file must be a TXT, DOC, PDF, JPEG or PNG';
+        const tooManyFilesError: string = 'You can only select up to 10 files at the same time';
+
         return {
             'upload-file': {
                 async handle(request: Request, response: Response): Promise<void> {
-                    await parseFormData(request, response);
-
-                    if (!request.file) {
-                        response.redirect(request.route.path);
-                        return;
-                    }
 
-                    const id = await that.fileTransferService.upload(request.file.buffer, request.file.originalname);
+                    const maxNumberOfFiles: number = Number(getEnvOrThrow('MAX_NUMBER_OF_FILES'));
```

Comment:
Throwing error in method bodies is not desired as it will fail only at the time users use the feature. Configuration should be validated at application start and fail fast. Please move it either outside of controller class or into constructor as last resort. Finally make sure application lunch crashes when value is missing.

Reply:
I put it outside the controller and it crashes